Nodejs/Product/Npm/SPI/NpmInstallCommand.cs

```diff
@@ -2,15 +2,20 @@
 
 namespace Microsoft.NodejsTools.Npm.SPI
 {
-    internal class NpmInstallCommand : NpmCommand
+    public class NpmInstallCommand : NpmCommand
     {
+        public static bool IsInstalling { get; private set; } = false;
+
         public NpmInstallCommand(
             string fullPathToRootPackageDirectory,
             string pathToNpm = null,
             bool useFallbackIfNpmNotFound = true)
             : base(fullPathToRootPackageDirectory, showConsole: false, pathToNpm: pathToNpm)
         {
             this.Arguments = "install";
+
+            this.CommandStarted += (sender, eventArgs) => { IsInstalling = true; };
```

Comment:
>CommandStarted [](start = 17, length = 14)

When is this unsubscribed?  Does it matter?

Reply:
It seems like it would be more natural for the consumer of the state changes to subscribe to the event, but I see that the event doesn't expose *which* command started.

---
In reply to: [206370382](https://github.com/Microsoft/nodejstools/pull/2015#discussion_r206370382) [](ancestors = 206370382,206369659)

Reply:
This is not unsubscribed, I don't think it matters tho.

I understood the `AbstractNpmLogSource` as a template class that provides a "hook" for when a command has started or finished. In consequence the `NpmInstallCommand` provides the implementation. 

The other place that might be a good option for this would be the class `NpmCommander`, which is actually the one that consumes the command. Seems like this is not the correct place to me though, as I think this should be just dumb and don't know the implementation details of the command.

Let me know what you think.